### PR TITLE
fix: catch SyntaxError on invalid JSON

### LINKED ISSUE #82: Catch SyntaxError and return 400 as well
In:

https://github.com/adobe/helix-universal/blob/85e54e625849ab54f5710ddd710340a9a11a90de/src/aws-adapter.js#L110-L121

we could additionally check whether `e` is of type `SyntaxError`, which would indicate that JSON parsing of the request body failed in `body-data-wrapper.js` (see https://github.com/adobe/helix-shared/issues/593)

Potential downside: some genuine parsing problem elsewhere in the code might be treated as bad request.

### PR DESCRIPTION
**Pull request by dominique-pfister**
fix #82

### CODE REVIEW

**Comment by dominique-pfister** [comment]
This fixes https://github.com/adobe/helix-shared/issues/593 as well

File: src/aws-adapter.js [hunk 1]
```diff
@@ -115,6 +115,16 @@ async function lambdaAdapter(event, context) {
...
       };
     }
+    if (e instanceof SyntaxError) {
+      return {
+        statusCode: 400,
+        headers: {
+          'content-type': 'text/plain',
+          'x-invocation-id': context.awsRequestId,
+        },
+        body: e.message,
```
Comment (dominique-pfister) [inline src/aws-adapter.js:125]: Might reveal too much, wdyt @tripodsan ?

**Comment by dominique-pfister** [comment]
And then, of course, if some JSON requested from a backend is malformed, and parsing is not guarded with a **try...catch** block, it would report a bad request, which might be confusing.

**Review by tripodsan — CHANGES_REQUESTED** [review]
I don't think this should be handled here.... if the `SyntaxError` is thrown where no input-data is involved, i.e. reading from a backend, a `500` would be more approriate.

**Comment by dominique-pfister** [comment]
Agree, so closing this

**Pull request closed without merging**